Library/ViewModels/PostCampaignCheckoutViewModel.swift

```diff
@@ -194,13 +203,16 @@ public class PostCampaignCheckoutViewModel: PostCampaignCheckoutViewModelType,
           .materialize()
       }
 
-    let paymentIntentClientSecret = newPaymentIntentEvent.values()
+    let paymentIntentClientSecretForExistingCards = newPaymentIntentForExistingCards.values()
       .map { $0.clientSecret }
 
     // Runs validation for pre-existing cards that were created with setup intents originally but require payment intents for late pledges.
     let validateCheckoutExistingCard = Signal
-      .combineLatest(checkoutId, selectedExistingCard, paymentIntentClientSecret, storedCardsValues)
+      .combineLatest(checkoutId, selectedCard, paymentIntentClientSecretForExistingCards, storedCardsValues)
```

Comment:
@ifosli Did I potentially just undo your bug fix from earlier?

Reply:
No, you're good - the important part is that we're looking for the most recently selected card in this line and that the filter to check if it's a new or existing card happens after (in line 213-215).